configurable default database, improved ad-hoc filters, parse key columns from query

vavrusa opened the pull request:
Hi, there are three improvements in this PR:

* Add configurable default database
* Allow wildcard ad-hoc filters for dashboards using multiple tables
* Parse dimensions from GROUP BY

Here's some rationale:

The datasource now allows default database to be configurable, this is used in two places:

* In query editor, the database field is prefilled with the default database
* In ad-hoc filters, the database is stripped from column names to make them
  less verbose (e.g. if default database is 'default', it will only show 'table.column' pairs)

The datasource also allows non-fully-qualified column names in the filter. In such case the database and table will be filled for each target independently, e.g. if there are two graphs in the dashboard with 'default.table1' and 'default.table2', and ad-hoc filter is 'column1 = 5', then it will apply in both cases. If the ad-hoc filter is 'table2.column1 = 5' then it will only apply in the second graph.

The columns representing dimensions are parsed from GROUP BY, which removes the need to wrap everything in subqueries with `groupArray()` and fixes several widget types (notably heatmap, table). If there's just time column, in the GROUP BY, it will use the metric alias name as before. If there's more
columns, it will generate the timeseries name as a join of dimension values, e.g: `GROUP BY t, OS, Country` can generate timeseries `Linux, US`

hagen1778 commented on src/sql_series.ts:
```diff
@@ -2,6 +2,7 @@ import _ from 'lodash';
 
 export default class SqlSeries {
     series: any;
+    keys: any;
```
Class `SqlSeries` also used in tests where it used within method `query` where `keys` are generated. To avoid tests failing we must add default `keys` value for `SqlSeries` objects [here](https://github.com/Vertamedia/clickhouse-grafana/blob/master/specs/datasource_specs.ts#L40) and [here](https://github.com/Vertamedia/clickhouse-grafana/blob/master/specs/datasource_specs.ts#L91).
Also it would be great if tests-specs will be populated with new tests.
I'm using `karma` to run tests:
`karma start -single-run`

vavrusa commented:
@hagen1778 Thanks for reviewing. I'm not sure how alerting is implemented, but it will need some sort of introspection anyway. I've fixed the remarks from this PR and added a short doc to README on how to build the project and run tests.

hagen1778 commented:
Thanks for contributing!

vavrusa commented:
@hagen1778 any chance of tagging this so we could install the module?

hagen1778 commented:
see https://github.com/Vertamedia/clickhouse-grafana/releases/tag/1.5.0

vavrusa commented:
@hagen1778 thanks!

vavrusa commented:
@hagen1778 I think you also need to bump the package version here https://github.com/Vertamedia/clickhouse-grafana/blob/master/package.json#L3 from 1.4.3 before submitting it to grafana plugin repo

hagen1778 commented:
Oh, you are totally right. That's my bad

hagen1778 commented:
Updated, thx again!
